features/storage/TESTS/blockdevice/general_block_device/main.cpp

```diff
@@ -420,6 +417,58 @@ void test_contiguous_erase_write_read()
     TEST_ASSERT_EQUAL(0, err);
 }
 
+void test_program_read_small_data_sizes()
+{
+    utest_printf("\nTest program-read small data sizes, from 1 to 7 bytes..\n");
+
+    BlockDevice *bd = BlockDevice::get_default_instance();
+
+    TEST_SKIP_UNLESS_MESSAGE(bd != NULL, "\nno block device found.\n");
+
+    // use BufferedBlockDevice for better handling of block devices program and read
+    BufferedBlockDevice *block_device = new BufferedBlockDevice(bd);
+
+    // BlockDevice initialization
+    int err = block_device->init();
+    TEST_ASSERT_EQUAL(0, err);
+
+    const char write_buffer[] = "1234567";
+    char read_buffer[7] = {};
+
+    bd_size_t erase_size = block_device->get_erase_size();
+    bd_size_t program_size = block_device->get_program_size();
+    TEST_ASSERT(program_size > 0);
+
+    // Determine starting address
+    bd_addr_t start_address = 0;
+
+    for (int i = 1; i < 7; i++) {
```

Comment:
Should be 'i < 8' if you want the test to run 7 times, to include the entire write_buffer "1234567"